### DISCUSSION TITLE
How to create L3 on Optimism L3

### DISCUSSION THREAD

**Discussion by gaurav-gupta** [ethereum-optimism/developers#407] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

We want to deploy L3 chain on Optimism L2 where gas would be paid in our ERC-20 Token. How to do that using OP-Stack or SuperChain. please help.

**Comment by opfocus** [comment]
>  I just got off a call with some protocol engineers. The primary hot spots of L3s are:
> 
> L(N) block time going from 12s → 2s is a big change:
> The L(N+1) needs to be less than L(N). If the L3 blocktime == L2 blocktime your chain will fail.
> The rollup node needs to be able to derive all the base chains' blocks to process deposit transactions.
> Any flakiness of that cross domain connection will cause the rollup to fall behind because the block times are on parity. It would never be able to catch up. Eventually it would fall behind the max sequencer drift and only produce empty blocks
> 
> Many of the configuration values are written in number of L1 blocks, so considerations need to be made their. You can generally multiple them by 6, but I don't have a working base config to reference.
> 
> EIP-1559 params assume a L3 block time of 2s. This is baked into the protocol design at the moment. The result is that fees can spike rapidly.
> 
> For plasmas: the bond size bakes in gas cost assumptions. (I don't have much detail on this)

Here are some quotes about L3. It may still not be enough for create L3.

**Reply by opfocus** [reply]
https://github.com/ethereum-optimism/developers/discussions/397

**Comment by Chomtana** [comment]
To deploy OP Stack as an L3, you can set the L1 RPC of your OP Stack to L2 RPC and you need to modify various polling intervals including

```
OP_NODE_L1_EPOCH_POLL_INTERVAL=2s
OP_NODE_L1_HTTP_POLL_INTERVAL=2s
```

But you may need to configure other values as well which I don't know yet. Please follow this discussion: https://github.com/ethereum-optimism/specs/discussions/158
